brainframe/cli/docker_compose.py

```diff
@@ -57,58 +66,48 @@ def run(install_path: Path, commands: List[str]) -> None:
 def download(target: Path, version: str = "latest") -> None:
     _assert_has_write_permissions(target.parent)
 
-    subdomain, auth_flags, version = check_download_version(version=version)
+    if version == "latest":
+        version = get_latest_version()
+
+    credentials = config.staging_credentials()
 
     url = BRAINFRAME_DOCKER_COMPOSE_URL.format(
-        subdomain=subdomain, version=version
-    )
-    os_utils.run(
-        ["curl", "-o", str(target), "--fail", "--location", url] + auth_flags,
+        subdomain="staging." if config.is_staging.value else "",
+        version=version,
     )
+    response = requests.get(url, auth=credentials, stream=True)
+    if not response.ok:
+        print_utils.fail_translate(
+            "general.error-downloading-docker-compose",
+            status_code=response.status_code,
+            error_message=response.text,
+        )
+
+    target.write_text(response.text)
 
     if os_utils.is_root():
         # Fix the permissions of the docker-compose.yml so that the BrainFrame
         # group can edit it
         os_utils.give_brainframe_group_rw_access([target])
 
 
-def check_download_version(
-    version: str = "latest",
-) -> Tuple[str, List[str], str]:
-    subdomain = ""
-    auth_flags = []
-
+def get_latest_version() -> str:
+    """
+    :return: The latest available version in the format "vX.Y.Z"
+    """
     # Add the flags to authenticate with staging if the user wants to download
     # from there
+    credentials: Optional[Tuple[str, str]]
     if config.is_staging.value:
         subdomain = "staging."
+    else:
+        subdomain = ""
+        credentials = None
 
-        username = config.staging_username.value
-        password = config.staging_password.value
-        if username is None or password is None:
-            print_utils.fail_translate(
-                "general.staging-missing-credentials",
-                username_env_var=config.staging_username.name,
-                password_env_var=config.staging_password.name,
-            )
-
-        auth_flags = ["--user", f"{username}:{password}"]
-
-    if version == "latest":
-        # Check what the latest version is
-        url = BRAINFRAME_LATEST_TAG_URL.format(subdomain=subdomain)
-        result = os_utils.run(
-            ["curl", "--fail", "-s", "--location", url] + auth_flags,
-            print_command=False,
-            stdout=subprocess.PIPE,
-            encoding="utf-8",
-        )
-        # stdout is a file-like object opened in text mode when the encoding
-        # argument is "utf-8"
-        stdout = cast(TextIO, result.stdout)
-        version = stdout.readline().strip()
-
-    return subdomain, auth_flags, version
+    # Check what the latest version is
+    url = BRAINFRAME_LATEST_TAG_URL.format(subdomain=subdomain)
```

Comment:
I noticed you did 

`subdomain="staging." if config.is_staging.value else ""`
elsewhere, and passed in the credentials regardless of whether they were needed. 

I'm just wondering if this call is different

Reply:
This code is just super broken lol. I never even load the credentials. Will feeeeeex.